core/trino-parser/src/main/java/io/trino/sql/jsonpath/PathTreeBuilder.java

```diff
@@ -0,0 +1,412 @@
+                return ComparisonPredicate.Operator.NOT_EQUAL;
+            case "<":
+                return ComparisonPredicate.Operator.LESS_THAN;
+            case ">":
+                return ComparisonPredicate.Operator.GREATER_THAN;
+            case "<=":
+                return ComparisonPredicate.Operator.LESS_THAN_OR_EQUAL;
+            case ">=":
+                return ComparisonPredicate.Operator.GREATER_THAN_OR_EQUAL;
+            default:
+                throw new UnsupportedOperationException("unexpected comparison operator: " + operator);
+        }
+    }
+
+    @Override
+    public PathNode visitConjunctionPredicate(JsonPathParser.ConjunctionPredicateContext context)
+    {
+        Predicate left = (Predicate) visit(context.left);
+        Predicate right = (Predicate) visit(context.right);
+        return new ConjunctionPredicate(left, right);
+    }
+
+    @Override
+    public PathNode visitDisjunctionPredicate(JsonPathParser.DisjunctionPredicateContext context)
+    {
+        Predicate left = (Predicate) visit(context.left);
+        Predicate right = (Predicate) visit(context.right);
+        return new DisjunctionPredicate(left, right);
+    }
+
+    @Override
+    public PathNode visitExistsPredicate(JsonPathParser.ExistsPredicateContext context)
+    {
+        PathNode path = visit(context.pathExpression());
+        return new ExistsPredicate(path);
+    }
+
+    @Override
+    public PathNode visitLikeRegexPredicate(JsonPathParser.LikeRegexPredicateContext context)
+    {
+        PathNode path = visit(context.base);
+        String pattern = unquote(context.pattern.getText());
+        Optional<String> flag = Optional.empty();
+        if (context.flag != null) {
+            flag = Optional.of(unquote(context.flag.getText()));
+        }
+        return new LikeRegexPredicate(path, pattern, flag);
+    }
+
+    @Override
+    public PathNode visitNegationPredicate(JsonPathParser.NegationPredicateContext context)
+    {
+        Predicate predicate = (Predicate) visit(context.delimitedPredicate());
+        return new NegationPredicate(predicate);
+    }
+
+    @Override
+    public PathNode visitParenthesizedPredicate(JsonPathParser.ParenthesizedPredicateContext context)
+    {
+        return visit(context.predicate());
+    }
+
+    @Override
+    public PathNode visitStartsWithPredicate(JsonPathParser.StartsWithPredicateContext context)
+    {
+        PathNode whole = visit(context.whole);
+        PathNode initial;
+        if (context.string != null) {
+            initial = visit(context.string);
+        }
+        else {
+            initial = namedVariable(context.NAMED_VARIABLE());
+        }
+        return new StartsWithPredicate(whole, initial);
+    }
+
+    @Override
+    public PathNode visitUnknownPredicate(JsonPathParser.UnknownPredicateContext context)
+    {
+        Predicate predicate = (Predicate) visit(context.predicate());
+        return new UnknownPredicate(predicate);
+    }
+}
```

Comment:
You may want to implement `aggregateResult` as in https://github.com/trinodb/trino/blob/edab967bedb09178ab0fa606ba77ca4223cbc23b/core/trino-parser/src/main/java/io/trino/sql/parser/AstBuilder.java#L3066-L3084, to prevent ast builder from skipping over syntax nodes it doesn't recognize